### PR TITLE
feat(adapter): update adapter to react17

### PR DESCRIPTION
**Pull request by Matthew-Mallimo**
## Description

Update jest adapter to react 17

## Motivation and Context

Prepare for one-app upgrade to react 17

## How Has This Been Tested?

unit, manual

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

**Comment by 10xLaCroixDrinker** [comment]
Make sure to add the BREAKING CHANGE note to the body of the commit.

**Review by 10xLaCroixDrinker — DISMISSED** [review]

**Comment by Matthew-Mallimo** [comment]
> Make sure to add the BREAKING CHANGE note to the body of the commit.

done

**Comment by 10xLaCroixDrinker** [comment]
Sorry for the run around but it needs to describe the breaking change. Like:

```
feat(adapter): update adapter to react 17

BREAKING CHANGE: drop support for react 16
```

**Review by 10xLaCroixDrinker — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -39,17 +39,16 @@
...
+    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.6",
     "amex-jest-preset": "^6.1.0",
-    "enzyme": "^3.0.0",
-    "enzyme-adapter-react-16": "^1.7.0",
+    "enzyme": "^3.11.0",
     "identity-obj-proxy": "^3.0.0"
   },
   "peerDependencies": {
     "enzyme-to-json": "^3.0.1",
     "jest": ">=24.0.0",
```
Comment (JAdshead) [inline package.json:49]: we should also update jest to 27.x

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Comment by Matthew-Mallimo** [comment]
Opening a new PR to avoid having to resolve conflicts

**Pull request closed without merging**